labs/02_integration/nackademin-app.postman_collection.json

```diff
@@ -1,98 +1,317 @@
+									"if (pm.response.code === 200 || pm.response.code === 201) {\r",
+									"  const p = pm.response.json();\r",
+									"  pm.environment.set(\"lastProductId\", p.id);\r",
+									"  pm.test(\"Has id\", () => pm.expect(p.id).to.exist);\r",
+									"}\r",
+									""
+								],
+								"type": "text/javascript",
+								"packages": {}
+							}
+						}
 					],
-					"path": [
-						"signup"
-					]
+					"request": {
+						"method": "POST",
+						"header": [
+							{
+								"key": "Authorization",
+								"value": "Bearer {{token}}",
+								"type": "text"
+							}
+						],
+						"body": {
+							"mode": "raw",
+							"raw": "{\r\n  \"name\": \"choklad\"\r\n}",
+							"options": {
+								"raw": {
+									"language": "json"
+								}
+							}
+						},
+						"url": {
+							"raw": "{{baseUrl}}/products",
+							"host": [
+								"{{baseUrl}}"
+							],
+							"path": [
+								"products"
+							]
+						}
+					},
+					"response": []
+				}
+			]
+		},
+		{
+			"name": "DEV",
+			"item": [
+				{
+					"name": "login_dev",
+					"event": [
+						{
+							"listen": "prerequest",
+							"script": {
+								"exec": [
+									""
+								],
+								"type": "text/javascript",
+								"packages": {}
+							}
+						},
+						{
+							"listen": "test",
+							"script": {
+								"exec": [
+									"// Kör bara om inloggningen lyckas\r",
+									"pm.test(\"Login responded\", () => pm.response.code > 0);\r",
+									"\r",
+									"if (pm.response.code === 200) {\r",
+									"  let data;\r",
+									"  try { data = pm.response.json(); } catch (e) { data = {}; }\r",
+									"\r",
+									"  // Stöd både access_token/token/jwt\r",
+									"  const tok = data.access_token || data.token || data.jwt;\r",
+									"\r",
+									"  // Sätt token i environment\r",
+									"  if (tok) {\r",
+									"    pm.environment.set(\"token\", tok);\r",
+									"  } else {\r",
+									"    pm.environment.unset(\"token\");\r",
+									"  }\r",
+									"\r",
+									"  // Tester\r",
+									"  pm.test(\"Token was received\", () => pm.expect(pm.environment.get(\"token\")).to.exist);\r",
+									"} else {\r",
+									"  // Misslyckad login → ta bort ev. gammal token så vi inte luras\r",
+									"  pm.environment.unset(\"token\");\r",
+									"  pm.test(`Login failed (${pm.response.code})`, () => pm.expect.fail(`Status ${pm.response.code}`));\r",
+									"}\r",
+									""
+								],
+								"type": "text/javascript",
+								"packages": {}
+							}
+						}
+					],
+					"request": {
+						"method": "POST",
+						"header": [
+							{
+								"key": "Authorization",
+								"value": "bearer {{token}}",
+								"type": "text"
+							}
+						],
+						"body": {
+							"mode": "raw",
+							"raw": "{\r\n  \"username\": \"admin_dev\",\r\n  \"password\": \"pass_1234\"\r\n}",
```

Comment:
This collection will not work towards different evironments because it has the user and password hardcoded instead of using postman environment variables.